### PR TITLE
Revisions and formatting for oracle commands

### PR DESCRIPTION
**Pull request by ajreynol**
It also makes changes to the explanation of oracle commands, and puts a placeholder for further explanation in the oracles section.

It also makes minor revisions to clean up the formatting of the oracle commands, and some minor changes concerning non-standard theories.

FYI @polgreen

### CODE REVIEW

File: language/sygus.tex [hunk 1]
```diff
@@ -569,7 +569,7 @@ \section{Semantics of Commands}
 and invoking a solver for this conjecture.
 This conjecture is of the form:
 \begin{alignat*}{1}
- & \exists f_1,\ldots,f_n.\, \forall v_1,\ldots,v_m.\,(\alpha_1 \wedge \ldots \wedge \alpha_r)[f_1,\ldots,f_n,v_1,\ldots,v_m] \implies (\varphi_1 \wedge \ldots \wedge \varphi_q )[f_1,\ldots,f_n,v_1,\ldots,v_m]
```
Comment (polgreen) [inline language/sygus.tex:572]: Why did you drop the `[f_1,\ldots,f_n,v_1,\ldots,v_m]`? 

It was originally like this (before I added the oracles), and I just extended this to include the assumptions:
~~~
\exists f_1,\ldots,f_n.\, \forall x_1,\ldots,x_m.\,(\varphi_1 \wedge \ldots \wedge \varphi_p )[f_1,\ldots,f_n,x_1,\ldots,x_m]
~~~
https://github.com/SyGuS-Org/docs/blame/6c71255f345d80080b6091b762a99c61c6b2df22/language/sygus.tex#L537

File: language/sygus.tex [hunk 2]
```diff
@@ -1046,79 +1042,74 @@ \subsection{Asserting Synthesis Constraints and Assumptions}
...
+For each such call,
+the formula $t[c_1 \ldots c_{n+m}]$, i.e., 
+$t$ with all occurences of $x_1, \ldots, x_{n+m}$ replaced with $c_1, \ldots, c_{n+m}$, 
+is added to the current list of constraints $\varphi$ in the conjecture.
+
+Note that 
+the synthesis solver may choose to call the binary $N$ 
+any time during solving, and as many times as it chooses.
+Note that it is possible that the synthesis solver may call the binary with
+the same input more than once.
```
Comment (polgreen) [inline language/sygus.tex:1079]: i agree with this change

File: language/sygus.tex [hunk 3]
```diff
@@ -1581,6 +1569,16 @@ \subsection{Input and Output Logics}
...
+\subsubsection{Other Theories}
+
+Many other theories
+are possible beyond those supported in the SMT-LIB standard.
+In this section, we mention theories that are of interest to synthesis applications
+that are not included in the SMT-LIB standard.
+
+\paragraph{Tables}
+TODO
+
```
Comment (polgreen) [inline language/sygus.tex:1581]: did you mean to include this in this PR?

File: language/sygus.tex (hunk 1 above)
Comment (polgreen) [reply language/sygus.tex:572]: This is the only change I have a real question about

**Review by polgreen — COMMENTED** [review]
Thanks for fixing my formatting. I have no strong opinions on parentheses, I'm not sure why I put the extra ones in and I'm happy with whatever you think is most consistent with the rest of SyGuS. 

The rewording looks fine to me

File: language/sygus.tex (hunk 3 above)
Comment (ajreynol) [reply language/sygus.tex:1581]: It's not critical, but yes, this was a dump of what my local changes were.  I could split this to another PR if you prefer.

File: language/sygus.tex (hunk 1 above)
Comment (ajreynol) [reply language/sygus.tex:572]: The reason why I'm changing now is simply an aesthetic one, that now the list of free variables in `varphi_1 ... varphi_p` is a bit large.

In general, the syntax `P[x]` is used to denote that the free variables of `P` are contained in `x`, this is implied by saying `\exists  \forall x. P` is a closed formula. 

I've made a commit to make this clearer, to say that the synthesis conjecture is a closed formula.

File: language/sygus.tex (hunk 3 above)
Comment (polgreen) [reply language/sygus.tex:1581]: I don't mind, fine as is

**Review by polgreen — APPROVED** [review]

**Pull request merged**